zilliqa/src/consensus.rs

```diff
@@ -635,6 +658,21 @@ impl Consensus {
             );
 
             let mut block_receipts: Vec<TransactionReceipt> = Vec::new();
+
+            if transactions.len() > 0 {
+                trace!("applying {} transactions to state", transactions.len());
+            }
+
+            // Must make sure state root hash is set to the parent's state root hash before applying transactions
```

Comment:
Previously forks were tolerated becaue most likely you would switch to a different branch and all the TXs would fail due to being executed before but the state hash would line up anyway.

Reply:
Yeah, that is fair, the check can go after the fork unwinding, then